core/src/services/gcs/core.rs

```diff
@@ -269,37 +276,28 @@ impl GcsCore {
 
         req = req.header(CONTENT_LENGTH, size.unwrap_or_default());
 
-        if metadata.is_empty() {
-            if let Some(content_type) = op.content_type() {
-                req = req.header(CONTENT_TYPE, content_type);
-            }
-
+        if request_metadata.is_empty() {
```

Comment:
I don't know if we should handle this case, or always use `uploadType=multipart`. It can happen  that the `request_metadata` serialization is an empty json (`{ }`) and this check does not use multipart if thats the case.

I'm not sure if it would be better to use always `uploadType=multipart`. I don't know if there are any problem at GCS if we send an empty metadata though... If the request can fail if the metadata is empty, then we must do this check

Reply:
`Multipart` involves additional handling of multipart encoding, so it's better to avoid those costs when users only want to write data.

Reply:
Agree with that!